lib/em-hiredis/pubsub_client.rb

```diff
@@ -248,10 +248,11 @@ def unsubscribe_impl(type, subscriptions, channel)
     end
 
     def unsubscribe_proc_impl(type, subscriptions, channel, proc)
-      removed = subscriptions[channel].delete(proc)
+      return unless subscriptions.include?(channel)
```

Comment:
Early return is the opposite pattern to all the other functions in this file, I'd prefer we were consistent.